store/iavl/proof.go

```diff
@@ -0,0 +1,122 @@
+package iavl
+
+import (
+	"errors"
+	"fmt"
+
+	ics23 "github.com/confio/ics23/go"
+	"github.com/tendermint/tendermint/crypto/merkle"
+)
+
+const ProofOpIAVLCommitment = "iavlstore"
```

Comment:
I would use a name like `ics23:iavl` as this is more clear and as we will be adding other `ics23:*` proofs that differ only in spec.